DSharpPlus/Net/Serialization/DiscordJson.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Linq;
+using System.Reflection;
+using DSharpPlus.Entities;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace DSharpPlus.Net.Serialization
+{
+    public static class DiscordJson
+    {
+        /// <summary>Serializes the specified object to a JSON string.</summary>
+        /// <param name="value">The object to serialize.</param>
+        /// <returns>A JSON string representation of the object.</returns>
+        public static string SerializeObject(object value)
+        {
+            return JsonConvert.SerializeObject(value, new OptionalJsonConverter());
+        }
+    }
+
+    public class OptionalJsonConverter : JsonConverter
+    {
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        {
+            var type = value.GetType();
+#if NETSTANDARD1_1 || NETSTANDARD1_3
```

Comment:
This is pretty unnecessary. Just make it a single code path. It will make maintenance and debugging easier.